src/JSON.ts

```diff
@@ -925,12 +943,76 @@ export class ParseResult {
         return this._commentTokens;
     }
 
+    public get commentTokenCount(): number {
+        return this._commentTokens.length;
+    }
+
     public get lineLengths(): number[] {
         return this._lineLengths;
     }
 
+    // Might or might not make a copy
+    public getTokens(commentBehavior: Comments): Token[] {
+        if (commentBehavior === Comments.includeCommentTokens) {
+            const tokens = this.tokens.concat(this.commentTokens);
+            tokens.sort((a, b) => a.span.startIndex - b.span.startIndex);
+            return tokens;
+        } else {
+            return this.tokens;
+        }
+    }
+
+    // Might or might not make a copy
+    public getTokensInSpan(span: language.Span, commentsBehavior: Comments): Token[] {
+        const results: Token[] = [];
+        const tokens = this.getTokens(commentsBehavior);
+        const spanStartIndex = span.startIndex;
+        const spanEndIndex = span.endIndex;
+
+        for (let token of tokens) {
+            if (token.span.endIndex >= spanStartIndex) {
+                if (token.span.startIndex > spanEndIndex) {
+                    break;
+                }
+
+                results.push(token);
+            }
+        }
+
+        return results;
+    }
+
+    public getLastTokenOnLine(line: number, commentBehavior: Comments = Comments.ignoreCommentTokens): Token | undefined {
+        const startOfLineIndex = this.getCharacterIndex(line, 0);
+        const lastLine = this.lineLengths.length - 1;
+
+        const tokens = this.getTokens(commentBehavior);
+
+        let lastSeenToken;
+
+        if (line === lastLine) {
+            // On last line, check the very last token
+            lastSeenToken = tokens[tokens.length - 1];
+        } else {
+            const nextLineIndex = this.getCharacterIndex(line + 1, 0);
+
+            for (let token of tokens) {
```

Comment:
>for (let token of tokens [](start = 12, length = 24)

Not sure how perf critical this is, but you could probably do a binary search here.

Reply:
Added comment, but probably not worth it right now.